sdk/identity/Azure.Identity/tests/VisualStudioCodeCredentialLiveTests.cs

```diff
@@ -97,6 +97,19 @@ public async Task AuthenticateWithVscCredential_TenantInSettings()
             Assert.IsNotNull(token.Token);
         }
 
+        [Test]
+        [RunOnlyOnPlatforms(Windows = true, OSX = true)]
+        public void AuthenticateWithVscCredential_NoVscInstalled()
+        {
+            var cloudName = Guid.NewGuid().ToString();
+
+            var fileSystem = CredentialTestHelpers.CreateFileSystemForVisualStudioCode(TestEnvironment, cloudName);
+            var options = Recording.InstrumentClientOptions(new VisualStudioCodeCredentialOptions { TenantId = TestEnvironment.TestTenantId });
+            VisualStudioCodeCredential credential = InstrumentClient(new VisualStudioCodeCredential(options, default, default, fileSystem, default));
+
+            Assert.CatchAsync<AuthenticationFailedException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[] {"https://vault.azure.net/.default"}), CancellationToken.None));
```

Comment:
This should be throwing `CredentialUnavailableException` no? While it does derive from `AuthenticationFailedException` we need to validate it explicitly to ensure the proper behavior when this credential is used in the `DefaultAzureCredential` or `ChainedTokenCredential`